### PR TITLE
Some tweaks

### PR DESCRIPTION
**Pull request by losvedir**
Each commit is self-contained, so might be easier to review that way.

/cc @kohlerjp

### CODE REVIEW

File: lib/gtfs_realtime_viz/state.ex [hunk 1]
```diff
@@ -31,22 +29,31 @@ defmodule GTFSRealtimeViz.State do
...
+      end
     end)
 
     {:reply, :ok, new_state}
   end
   def handle_call({:vehicles, group}, _from, state) do
     {:reply, state[group], state}
   end
+
+  defp max_archive, do: Application.get_env(:gtfs_realtime_viz, :max_archive)
```
Comment (kohlerjp) [inline lib/gtfs_realtime_viz/state.ex]: What about using a default argument for the `get_env` call? That way there's no chance of calling `Enum.take(enum, nil)` if the parent application doesn't set it. Maybe that could also be useful for the `routes` config above.

File: lib/gtfs_realtime_viz/state.ex [hunk 2]
```diff
@@ -31,22 +29,31 @@ defmodule GTFSRealtimeViz.State do
...
   # server callbacks
 
   def handle_call({:vehicles, group, vehicles, comment}, _from, state) do
     new_state = update_in(state, [Access.key(group, [])], fn prev_msgs ->
-      Enum.take([{comment, vehicles} | prev_msgs], @max_archive)
+      msgs = [{comment, vehicles} | prev_msgs]
+      if max_archive() == :infinity do
+        msgs
+      else
+        Enum.take(msgs, max_archive())
```
Comment (kohlerjp) [inline lib/gtfs_realtime_viz/state.ex]: Small thing, but you could set the value of `max_archive()` to a variable so the calculation only needs to be made at most once, even though it's a pretty low stakes operation.

Comment (losvedir) [reply lib/gtfs_realtime_viz/state.ex]: Sure, good idea!

File: lib/gtfs_realtime_viz/state.ex (hunk 1 above)
Comment (losvedir) [reply lib/gtfs_realtime_viz/state.ex]: There's a default of `5` already provided in `mix.exs`:

```ex
  def application do
    [
      extra_applications: [:logger],
      mod: {GTFSRealtimeViz.Application, []},
      env: env(),
    ]
  end

  defp env do
    [
      max_archive: 5,
      routes: %{},
    ]
  end
```

So the only way it'd be `nil` is... I'm not sure... maybe if someone manually deleted it from the environment (not sure if that's possible). In that case, I feel like we shouldn't provide a backup here.

Comment (kohlerjp) [reply lib/gtfs_realtime_viz/state.ex]: Hmm, I'm seeing that error when I don't set the config in the RTR application. Does this env still get called as a child application? Or does that just mean I'm using the wrong commit of this app

Comment (losvedir) [reply lib/gtfs_realtime_viz/state.ex]: I can't reproduce... can you try this again? I tried this:

1. Pull recent `master` for `rtr`.
2. Delete the `config :gtfs_realtime_viz, :max_archive, :infinity` line
3. `iex -S mix`
4. In iex do `GTFSRealtimeViz.new_message(:prod, File.read!("/Users/gdurazo/Downloads/VehiclePositions.pb"), "test")` a bunch of times
5. `GTFSRealtimeViz.State.vehicles(:prod)` - it kept 5 messages (the default in the viz repo).
6. Add back that config line, do `iex -S mix` again
7. Send more messages, and `State` shows that all have been kept now.

Seemed to work as expected.

**Pull request merged**